From d780d3b779bd0ec65f75f4119bc7bdf3bf4aa3ac Mon Sep 17 00:00:00 2001
From: Aubrey Li <aubrey.li@intel.com>
Date: Sun, 11 May 2014 05:47:58 -0500
Subject: [PATCH 088/152] Add PMC device state report for suspend debug
 purpose

S0ix requires all of the devices under south complex are in D3, we
can find the device states report in dmesg if pm_suspend_dev_state()
is called after suspend
---
 kernel/power/suspend.c |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

--- linux.orig/kernel/power/suspend.c
+++ linux/kernel/power/suspend.c
@@ -38,6 +38,24 @@ const char *const pm_states[PM_SUSPEND_M
 
 static const struct platform_suspend_ops *suspend_ops;
 
+void (*sc_dev_state)(void *);
+EXPORT_SYMBOL(sc_dev_state);
+void (*nc_dev_state)(void *);
+EXPORT_SYMBOL(nc_dev_state);
+
+void pm_suspend_dev_state(void)
+{
+	if (!pm_suspend_debug)
+		return;
+	else {
+		if (sc_dev_state)
+			sc_dev_state(NULL);
+		if (nc_dev_state)
+			nc_dev_state(NULL);
+	}
+}
+EXPORT_SYMBOL(pm_suspend_dev_state);
+
 static bool need_suspend_ops(suspend_state_t state)
 {
 	return !!(state > PM_SUSPEND_FREEZE);
